src/dsp_tools/utils/xml_upload_stash.py

```diff
@@ -65,12 +62,34 @@ def _log_unable_to_upload_xml_resource(
     logger.warning(err_msg, exc_info=True)
 
 
+def _log_iri_does_not_exist_error(
+    received_error: KeyError,
+    stashed_resource: XMLResource,
+    all_link_props: XMLProperty,
+) -> None:
+    """
+    This function logs if it is not possible to upload an XML resource
+    if a linked resource does not have an IRI in the triplestore.
+
+    Args:
+        received_error: Error received
+        stashed_resource: resource that is stashed
+        all_link_props: all the link properties from that resource
+    """
+    err_msg = (
+        f"Unable to upload the xml text of '{all_link_props.name}' of resource '{stashed_resource.id}'"
+        f"because the resource with the internal id '{received_error.args[0]}' does not exist in the triplestore."
```

Comment:
I find the pattern try - except - log error more complicated and more verbose than the previously used pattern:

```
try:
    my_val = my_dict["foo"]
except KeyError as kerr:
    _log_key_error(kerr)

def _log_key_error(kerr: KeyError):
    logger.warning(f"The error kerr.args[0] occurred!")
```

```
if my_val not in my_dict:
    _log_that_resource_doesnt_exist("The resource doesn't exist")
my_val = my_dict["foo"]

def _log_that_resource_doesnt_exist(msg: str):
    logger.warning(msg)
```

In addition, passing a string to a function is more compliant to the Law of Demeter (=Principle of Least Knowledge) than passing a KeyError to it. Because in fact you don't need the entire KeyError, you only need the ID that was not found in the dictionary. It would be much easier to pass the failing ID as string, than passing a KeyError and then extracting the string with received_error.args[0].

Because

Reply:
1. Regarding: try - except - log error
I wrote it that way because a try, except is faster than an if clause if it occurs less than 50% of the time. Which I prioritized here since performance time is our focus here.

2. Regarding the KeyError Logging
This I can understand and can fix easily.